src/main/java/io/radanalytics/operator/cluster/KubernetesSparkClusterDeployer.java

```diff
@@ -26,10 +26,16 @@
     public KubernetesResourceList getResourceList(SparkCluster cluster) {
         synchronized (this.client) {
             String name = cluster.getName();
+
+            Map<String, String> allMasterLabels = new HashMap<>();
+            if (cluster.getLabels() != null) allMasterLabels.putAll(cluster.getLabels());
+            if (cluster.getMaster() != null && cluster.getMaster().getLabels() != null)
+                allMasterLabels.putAll(cluster.getMaster().getLabels());
```

Comment:
yeah, this makes sense. Master/worker specific labels override the global ones :+1: